packages/block-library/src/query/index.php

```diff
@@ -31,8 +31,16 @@ function render_block_core_query( $attributes, $content, $block ) {
 			// Add the necessary directives.
 			$p->set_attribute( 'data-wp-interactive', 'core/query' );
 			$p->set_attribute( 'data-wp-router-region', 'query-' . $attributes['queryId'] );
-			$p->set_attribute( 'data-wp-context', '{}' );
+			$p->set_attribute(
+				'data-wp-context',
+				wp_json_encode(
+					array(
+						'isCurrentNavigationOrigin' => false,
+					)
```

Comment:
Perhaps we could add a small comment about how to use it?

```suggestion
					array(
						/**
						 * Checks whether the current navigation was originated from this Query block.
						 *
						 * Usage:
						 *   ```js
						 *   const { isCurrentNavigationOrigin } = getContext( 'core/query' );
						 *   ```
						 */
						'isCurrentNavigationOrigin' => false,
					)
```